zebrad/tests/acceptance.rs

```diff
@@ -9,36 +9,50 @@ use std::time::Duration;
 use zebra_test::prelude::*;
 
 // Todo: The following 3 helper functions can probably be abstracted into one
-fn get_child_single_arg(arg: &str) -> Result<zebra_test::command::TestChild> {
-    let (mut cmd, _guard) = test_cmd(env!("CARGO_BIN_EXE_zebrad"))?;
-
-    cmd.arg(arg)
-        .stdout(Stdio::piped())
-        .stderr(Stdio::piped())
-        .spawn2()
+fn get_child_single_arg(arg: &str) -> Result<(zebra_test::command::TestChild, impl Drop)> {
+    let (mut cmd, guard) = test_cmd(env!("CARGO_BIN_EXE_zebrad"))?;
+
+    Ok((
+        cmd.arg(arg)
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn2()
+            .unwrap(),
+        guard,
+    ))
 }
 
-fn get_child_multi_args(args: &[&str]) -> Result<zebra_test::command::TestChild> {
-    let (mut cmd, _guard) = test_cmd(env!("CARGO_BIN_EXE_zebrad"))?;
-
-    cmd.args(args)
-        .stdout(Stdio::piped())
-        .stderr(Stdio::piped())
-        .spawn2()
+fn get_child_multi_args(args: &[&str]) -> Result<(zebra_test::command::TestChild, impl Drop)> {
+    let (mut cmd, guard) = test_cmd(env!("CARGO_BIN_EXE_zebrad"))?;
+
+    Ok((
+        cmd.args(args)
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn2()
+            .unwrap(),
+        guard,
+    ))
 }
 
-fn get_child_no_args() -> Result<zebra_test::command::TestChild> {
-    let (mut cmd, _guard) = test_cmd(env!("CARGO_BIN_EXE_zebrad"))?;
+fn get_child_no_args() -> Result<(zebra_test::command::TestChild, impl Drop)> {
+    let (mut cmd, guard) = test_cmd(env!("CARGO_BIN_EXE_zebrad"))?;
 
-    cmd.stdout(Stdio::piped()).stderr(Stdio::piped()).spawn2()
+    Ok((
+        cmd.stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn2()
+            .unwrap(),
+        guard,
+    ))
 }
 
 #[test]
 fn generate_no_args() -> Result<()> {
     zebra_test::init();
 
     let child = get_child_single_arg("generate");
-    let output = child.unwrap().wait_with_output()?;
+    let output = child.unwrap().0.wait_with_output()?;
```

Comment:
Why'd you switch to waiting on the inner `Child` here? This will not add context to the error report if something goes wrong, such as what the original command was.

Reply:
Thanks.